### PR TITLE
Add KrnlSqrtOp

### PR DESCRIPTION
**Pull request by tungld**
Some ONNX operators like: `Sqrt` or `BatchNormalization` require to compute square root, but there is no `SqrtOp` in the standard dialect in MLIR. 

This patch is to implement a KrnlSqrtOp in the Krnl dialect and lower it to LLVM. ONNX operators can be lowered to this KrnlSqrtOp.

### CODE REVIEW

File: src/dialect/krnl/krnl_ops.td [hunk 1]
```diff
@@ -190,3 +190,17 @@ def KrnlMemcpyOp : Op<Krnl_Dialect, "memcpy"> {
...
+
+def KrnlSqrtOp : Op<Krnl_Dialect, "sqrt", [NoSideEffect]> {
+  let summary = "Krnl sqrt operation";
+  let description = [{
+    "The `sqrt` computes the square root value. It takes one operand and returns
+    one result with the same type."
+  }];
+
+  let arguments = (ins FloatLike:$operand);
+  let results = (outs AnyType);
```
Comment (doru1004) [inline src/dialect/krnl/krnl_ops.td]: Why not FloatLike ?

File: test/mlir/krnl/sqrt.mlir [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// RUN: onnf-opt --shape-inference --lower-all-llvm %s -split-input-file | FileCheck %s
+
+func @test_sqrt(%arg0 : f32) -> f32 {
+  %0 = "krnl.sqrt"(%arg0) : (f32) -> f32
```
Comment (doru1004) [inline test/mlir/krnl/sqrt.mlir]: Can you add another case with f64?

**Comment by doru1004** [comment]
So does this have to be done for other math functions also? Do we need to have a special dialect for this types of operations or is the standard MLIR dialect slowly adding these types of operations? If this is the case I would advocate for submitting a patch to MLIR to add SQRT there. Let me know your thoughts.

**Comment by tungld** [comment]
Some other math functions, like sin, tan, pow, should be done too. I am not sure about the standard MLIR dialect's future plan, but it is a good idea to port their lowing to MLIR. I will submit a patch to MLIR.

For our ONNF, if we'd like to run ResNet50, I recommend to support Sqrt soon since we need it for BatchNormalization.

File: test/mlir/krnl/sqrt.mlir (hunk 2 above)
Comment (tungld) [reply test/mlir/krnl/sqrt.mlir]: Done.

File: src/dialect/krnl/krnl_ops.td (hunk 1 above)
Comment (tungld) [reply src/dialect/krnl/krnl_ops.td]: I changed it. Thanks!

**Comment by tungld** [comment]
In order to check correctness of KrnlSqrtOp, I lowered ONNXSqrtOp to KrnlSqrtOp and added e2e tests.

**Pull request merged**